dev-docs/howtoguides/how_to_setup_windows_machine_for_wsl_testing.md

```diff
@@ -0,0 +1,158 @@
+
+Set an username and password for the VM. Please set the username as *ubuntu*. 
+We will use this account setting to manually login into the machine to run some commands there directly.
+![windows_secuirty_type](windows-ssh-setup.png)
+
+Finally, remember to also select the SSH inbound port for the machine
+![windows_secuirty_type](windows-setup-port.png)
+
+
+## Configure Bastion to login into the machine
+
+To login into the machine and access its GUI, you need to setup Bastion.
+After launching the machine, go for the *Connect with Bastion* option:
+![windows_secuirty_type](windows-bastion.png)
+![windows_secuirty_type](windows-deploy-bastion.png)
+
+After deploying Bastion, you just need to present your username and password to login into the
+machine.
+
+## Configuring Hyper-V
+
+After login into the machine through Bastion, you need to configure Hyper-V by following
+this [tutorial](https://learn.microsoft.com/en-us/azure/lab-services/how-to-enable-nested-virtualization-template-vm-using-script?tabs=powershell)
+
+## Installing winget
+
+For each Ubuntu distro, we will get the installers directly from the Microsoft store.
+To install such installers, we need to configure *winget*.
+
+To install *winget* on the machine, follow this [tutorial](https://winget.pro/winget-install-powershell/)
+
+```{note}
+if you see the following error when installing *winget*, restart the machine first:
+
+Add-AppxPackage : Deployment failed with HRESULT: 0x80073D02, The package could not be installed
+because resources it   modifies are currently in use. 
+```
+
+## Installing wsl
+
+To install *wsl*, just run the following command, as an administrator, on powershell:
+
+```console
+$ wsl --install
+```
+
+After that, we need to update WSL to the pre-release version by running:
+
+```console
+$ wsl --update --pre-release
+```
+
+Now, install a Bionic instance to test the whole WSL setup is working as expected:
+
+```console
+$ winget install --id "9PNKSF5ZN4SW" --accept-source-agreements --accept-package-agreements --silent
+```
+
+Now, create the image with:
+
+```console
+$ & ubuntu1804.exe install --root --ui=none
+```
+
+After that, just test to see if we can run commands on the WSL instance:
+
+```console
+$ wsl -d Ubuntu-18.04 --exec lsb_release -a
+```
+
+## Installing OpenSSH server
+
+To install OpenSSH server, follow these steps:
+
+1) Open *Settings*
+2) Go to *Apps*
+3) Go to *Optional Features*
+4) Go to *Add an optional feature*
+5) Search for *ssh*
+6) Install OpenSSH Server
+
+After installing the Service, open PowerShell and start-it by running:
+
+```console
+$ Start-Service sshd
+```
+
+And set it to automatically start after boot:
+
+```console
+$ Set-Service -Name sshd -StartupType 'Automatic'
+```
+
+### Configure the SSH keys
+
+We need to add a SSH public key into the Windows machine.
+
+To do that, follow these steps:
+
+1) Send the public key to the Windows machine:
+
+```console
+$ scp PUB_KEY_PATH ubuntu@IP:%programdata%/ssh
```

Comment:
```suggestion
$ scp PUB_KEY_PATH ubuntu@$WINDOWS_MACHINE_IP_ADDRESS:%programdata%/ssh
```
to make it more obvious that IP needs to be replaced

Reply:
or maybe explain the  `UACLIENT_BEHAVE_WSL_IP_ADDRESS` environment variable up here and use that

Reply:
I'd suggest both